plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -20,9 +20,25 @@ def initialize(job, output)
       @reference = job.deploy.reference
     end
 
+    def preview(resolve_build: true)
+      verify_kubernetes_templates!
+      @release = build_release(resolve_build: resolve_build)
+      unless @release.valid?
+        raise Samson::Hooks::UserError, "Failed to store manifests: #{release.errors.full_messages.inspect}"
+      end
+
+      @release.release_docs
+    end
+
     def execute(*)
       verify_kubernetes_templates!
-      @release = create_release
+      @release = build_release
+
+      Kubernetes::Release.transaction do
+        unless @release.save
+          raise Samson::Hooks::UserError, "Failed to store manifests: #{release.errors.full_messages.inspect}"
+        end
+      end
```

Comment:
these should be some kind of method maybe and then be reused .. looks very similar

Reply:
similarity is pretty small, prefer to leave it for now